spec/integration/util/execution_spec.rb

```diff
@@ -52,4 +52,24 @@
       expect(result.exitstatus).to eq(100)
     end
   end
+
+  describe "#execute (stdin_string)", :if => Puppet.features.posix? do
+    it "should write stdin_string to stdin of the process" do
+      input = 'test string'
+      output = Puppet::Util::Execution.execute('cat', :stdin_string => input, :combine => true)
+      expect(output).to eq(input)
+    end
+  end
+
+  describe "#execute (stdin_yield)", :if => Puppet.features.posix? do
+    it "should connect stdin and stdout to the process" do
+      input = 'test string'
+      output = Puppet::Util::Execution.execute('cat', :stdin_yield => true, :combine => true) do |i,o,e|
+        i.write(input)
+        i.close
+        o.read
+      end
+      expect(output).to eq(input)
+    end
+  end
```

Comment:
What is the behavior on Windows?

Reply:
I don't have a Windows machine to test with directly, and I previously (back in June) was unable to get the appveyor tests to pass for early revisions of this PR, hence this posix-only code. However, the tests are passing now, so I've updated everything in this PR to support Windows.